traffic_ops/traffic_ops_golang/staticdnsentry/staticdnsentry.go

```diff
@@ -81,13 +81,27 @@ func (staticDNSEntry *TOStaticDNSEntry) SetKeys(keys map[string]interface{}) {
 
 // Validate fulfills the api.Validator interface
 func (staticDNSEntry TOStaticDNSEntry) Validate() error {
-	if _, err := tc.ValidateTypeID(staticDNSEntry.ReqInfo.Tx.Tx, &staticDNSEntry.TypeID, "staticdnsentry"); err != nil {
+	typeStr, err := tc.ValidateTypeID(staticDNSEntry.ReqInfo.Tx.Tx, &staticDNSEntry.TypeID, "staticdnsentry");
+	if err != nil {
 		return err
 	}
 
+	addressErr := error(nil)
```

Comment:
this part seems odd to be in this PR. was this the result of a merge or rebase or something?

Reply:
this is a conditional validation for the 'address' field of the static dns entry form: a static dns entry with type A_RECORD needs to be a valid IPv4 address, AAAA_RECORD needs to be a valid IPv6 address, and CNAME_RECORD needs to be a valid DNS name. I just added the enhanced validation as part of the front end changes, since it was related.

Reply:
oh i see

Reply:
also,  it's not necessary to initialize that to nil,  since it's initialized by default -- just declare it:
```
var addressErr error
```

Reply:
>also, it's not necessary to initialize that to nil, since it's initialized by default -- just declare it:
> var addressErr error

I don't agree - I prefer to declare/initialize all variables with the same syntax in Go, `:=`, for consistency. I don't -1 people who use `var`, but I'd rather we didn't demand its usage, for people (like me) who prefer a single initialization syntax everywhere.